moonstone/analysis/diversity/base.py

```diff
@@ -168,12 +172,37 @@ def analyse_groups(
         :param output_file: file path to output your html graph
         :param make_graph: whether or not to make the graph
         :param plotting_options: plotly plotting_options
+        :param stats_test: {'mann_whitney_u', 'ttest_independence', 'chi2_contingency'} statistical test
+        used to calculate the p-values between each groups
+        :param correction_method: {None, 'fdr_bh' (benjamini-hochberg), 'bonferoni'} method used (if any)
+        to correct generated p-values
+        :param output_pvalue: {'series', 'dataframe'}
+        :param sym: whether generated dataframe (or MultiIndexed series) is symetric or half-full
         """
         filtered_metadata_df = self._get_filtered_df_from_metadata(metadata_df)
         df = self._get_grouped_df(filtered_metadata_df[group_col])
-        pval = statistical_test_groups_comparison(
-            df[self.DIVERSITY_INDEXES_NAME], df[group_col], stats_test
-        )
+
+        if correction_method is not None:
```

Comment:
What you can do here to make the code easier to read and avoid long method is to refactor all of your code into a method of the class:

```python
df = self._get_grouped_df(filtered_metadata_df[group_col])
```

here you would get roughly something like:

```python
pval = self._run_statistical_test_groups(stats_test, correction_method, sym, output_pvalue)
```

and have you method

```python
def self _run_statistical_test_groups(self, stats_test: str, correction_method: str, sym: bool, output_pvalue: str):
    #your code
```

Therefore you only give what's necessary to perform the action and it gets much easier to debug but also test.